<?xml version="1.0"?>
<launch>

  <arg name="subToR1Pose" default="True"/>
  <arg name="subToR2Pose" default="True"/>
  <arg name="subToR3Pose" default="True"/>
  <arg name="subToR4Pose" default="True"/>
  <arg name="subToR5Pose" default="True"/>

  <include file="$(find mocap_optitrack)/launch/mocap.launch"/>

  <param name = "subToR1Pose" value = "$(arg subToR1Pose)" />
  <param name = "subToR2Pose" value = "$(arg subToR2Pose)" />
  <param name = "subToR3Pose" value = "$(arg subToR3Pose)" />
  <param name = "subToR4Pose" value = "$(arg subToR4Pose)" />
  <param name = "subToR5Pose" value = "$(arg subToR5Pose)" />

  <!-- Robot_1 -->
  <group ns="Robot_1" if="$(eval subToR1Pose)">
    <rosparam param="waypoints_x">[3.9]</rosparam>
    <rosparam param="waypoints_y">[0.225]</rosparam>
    <rosparam param="waypoints_yaw">[-1.57]</rosparam>

  </group>

  <!-- Robot_2 -->
  <group ns="Robot_2" if="$(eval subToR2Pose)">
    <rosparam param="waypoints_x">[3.9]</rosparam>
    <rosparam param="waypoints_y">[0.6]</rosparam>
    <rosparam param="waypoints_yaw">[-1.57]</rosparam>
  </group>

  <!-- Robot_3 -->
  <group ns="Robot_3" if="$(eval subToR3Pose)">
    <rosparam param="waypoints_x">[3.9]</rosparam>
    <rosparam param="waypoints_y">[0.975]</rosparam>
    <rosparam param="waypoints_yaw">[-1.57]</rosparam>
  </group>

  <!-- Robot_4 -->
  <group ns="Robot_4" if="$(eval subToR4Pose)">
    <rosparam param="waypoints_x">[3.9]</rosparam>
    <rosparam param="waypoints_y">[1.35]</rosparam>
    <rosparam param="waypoints_yaw">[-1.57]</rosparam>
  </group>

  <!-- Robot_5 -->
  <group ns="Robot_5" if="$(eval subToR5Pose)">
    <rosparam param="waypoints_x">[3.9]</rosparam>
    <rosparam param="waypoints_y">[1.725]</rosparam>
    <rosparam param="waypoints_yaw">[-1.57]</rosparam>
  </group>

</launch>
